### DISCUSSION TITLE
How to set an element of a SelectionList or any widget as highlighted both logicaly and visually

### DISCUSSION THREAD

**Discussion by montaropdf** [Textualize/textual#5107] [Q&A]
Hello,

I have a screen containing a `SelectionList` and 2 `Button` widgets. Those buttons allows to move up or down the currently highlighted element of the `SelectionList`. However, once clicking on one of the button, no element is highlighted in the `SelectionList`.

I have tried to give back the focus to the `SelectionList` with the `.focus()` method at the end of the method triggered when pressing a button, but, it does not highlight any element of the widget.

I use a SelectionList instead of a ListView, because the cited widgets, above, are part of a more complex screen and a SelectionList makes more sense for the purpose of the app. However, the same problem would arise with any "highlightable" widget, I suppose.

**Comment by TomJGooding** [comment]
When the `SelectionList` isn't focused, the highlighted item is only styled with bold text without the background change. Is that what you mean? You could add something like this to your CSS to change that: 

```css
    SelectionList {
        & > .option-list--option-highlighted {
            background: $accent;
        }
```

**Reply by montaropdf** [reply]
What I would like is
1. give the focus to the option that was highlighted before clicking on the button
2. highlight it.

**Reply by TomJGooding** [reply]
When you click the button, that widget will now have focus. If you want to move focus back to the selection list after the button is pressed, something like `selection_list.focus()` should work. Here's a quick example:

```python
from textual import on
from textual.app import App, ComposeResult
from textual.containers import Grid
from textual.widgets import Button, SelectionList


class ExampleApp(App):
    CSS = """
    Grid {
        height: auto;
        grid-size: 2;
    }

    SelectionList {
        row-span: 2;
    }
    """

    def compose(self) -> ComposeResult:
        with Grid():
            yield SelectionList[int](
                ("One", 1),
                ("Two", 2),
                ("Three", 3),
                ("Four", 4),
            )
            yield Button("Up", id="up-btn")
            yield Button("Down", id="down-btn")

    @on(Button.Pressed, "#up-btn")
    def move_selection_up(self) -> None:
        selection_list = self.query_one(SelectionList)
        selection_list.action_cursor_up()
        selection_list.focus()

    @on(Button.Pressed, "#down-btn")
    def move_selection_down(self) -> None:
        selection_list = self.query_one(SelectionList)
        selection_list.action_cursor_down()
        selection_list.focus()


if __name__ == "__main__":
    ExampleApp().run()
```

**Comment by montaropdf** [comment]
ok, I will test asap, I am a bit busy at the moment.
